src/Service/ProgrammesAggregationService.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Service;
+
+use BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository\ClipRepository;
+use BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository\CoreEntityRepository;
+use BBC\ProgrammesPagesService\Domain\Entity\Programme;
+use BBC\ProgrammesPagesService\Mapper\ProgrammesDbToDomain\CoreEntityMapper;
+use BBC\ProgrammesPagesService\Cache\CacheInterface;
+use InvalidArgumentException;
+
+class ProgrammesAggregationService extends AbstractService
+{
+    const AGGREGATION_PROGRAMMES_VALID_TYPES = [
+        'Series',
+        'Episode',
+        'Clip',
+    ];
+
+    public function __construct(
+        CoreEntityRepository $repository,
+        CoreEntityMapper $mapper,
+        CacheInterface $cache
+    ) {
+        parent::__construct($repository, $mapper, $cache);
+    }
+
+    /**
+     * @return Programme[]
+     */
+    public function findProgrammesChildrenByType(
```

Comment:
How about `findDescendantClips` - then other filters can have separate names `findDescendantEpisodes` etc. That shall help discoverability in your IDE as your function names make it clear what things do.


Note that this is in contrast o the Repository functions - the Repositories are private - they don't need to be accessed outside of this library. This means that we should focus on reducing the number of function by reusing / finding common ground where appropriate, rather than developer experience.

Services - public - DX means discoverability by IDE hints on function names and not having to guess parameter values.
Repositories - private - DX means keeping a small but flexible footprint when making things that the Services use.